### PR TITLE
ns-plug: added enterprise distfeed

### PR DESCRIPTION
**Pull request by Tbaile**
Ref: #412

### CODE REVIEW

File: packages/ns-plug/files/distfeed-setup [hunk 1]
```diff
@@ -5,15 +5,25 @@
 # SPDX-License-Identifier: GPL-2.0-only
 #
 
-base_url=${BASE_URL:='https://updates.nethsecurity.nethserver.org'}
+base_url=${BASE_URL:=''}
```
Comment (gsanchietti) [inline packages/ns-plug/files/distfeed-setup:8]: We could totally remove this line

Comment (Tbaile) [reply packages/ns-plug/files/distfeed-setup:8]: Dunno what scripts actually use the manual env set of `BASE_URL`, left it there just in case :shrug:

Comment (gsanchietti) [reply packages/ns-plug/files/distfeed-setup:8]: we can just remove the support of env var
```suggestion
base_url="$(uci -q get ns-plug.config.repository_url)"
```

File: packages/ns-plug/files/distfeed-setup [hunk 2]
```diff
@@ -5,24 +5,27 @@
...
 channel=${CHANNEL:=''}
 owrt_version=${OWRT_VERSION:=''}
 
+if [ -z "$base_url" ]; then
+    base_url="$(uci -q get ns-plug.config.repository_url)"
+fi
+
```
Comment (gsanchietti) [inline packages/ns-plug/files/distfeed-setup:15]: Not needed anymore if initialized on top.
```suggestion
```

File: packages/ns-api/files/ns.update [hunk 3]
```diff
@@ -86,11 +87,15 @@ def install_package_updates():
     return {"result": "success"}
 
 def check_system_update():
+    e_uci = EUci()
     current_version = get_system_version()
     ret = {"currentVersion": f'NethSecurity {current_version}', "lastVersion": "", "scheduledAt": get_update_schedule()}
     try:
-        channel = get_distfeed_channel()
-        with urllib.request.urlopen(f"https://updates.nethsecurity.nethserver.org/{channel}/latest_release") as resp:
+        url = f"{e_uci.get('ns-plug', 'config', 'repository_url')}"
```
Comment (gsanchietti) [inline packages/ns-api/files/ns.update:94]: No need to cast to string:
```suggestion
        url = e_uci.get('ns-plug', 'config', 'repository_url', default="https://updates.nethsecurity.nethserver.org")
```

**Review by gsanchietti — COMMENTED** [review]
Also add default repository_url to https://github.com/NethServer/nethsecurity/blob/main/packages/ns-plug/files/config

**Comment by gsanchietti** [comment]
Also, for the HTTP basic authentication take a look at this: https://stackoverflow.com/questions/2407126/python-urllib2-basic-auth-problem

File: packages/ns-api/files/ns.update (hunk 3 above)
Comment (Tbaile) [reply packages/ns-api/files/ns.update:94]: I'd remove the default, it is set anyway, by default pyuci returns a string type
```suggestion
        url = e_uci.get('ns-plug', 'config', 'repository_url')
```

Comment (gsanchietti) [reply packages/ns-api/files/ns.update:94]: ok remove the default and the f-string

File: packages/ns-plug/files/register [hunk 4]
```diff
@@ -76,6 +76,8 @@ esac
 
 uci set ns-plug.config.system_id="$system_id"
 uci set ns-plug.config.secret="$secret"
+uci set ns-plug.config.repository_url="https://$system_id:$secret@distfeed.nethesis.it/repository/$type/nethsecurity"
+
```
Comment (gsanchietti) [inline packages/ns-plug/files/register:80]:
```suggestion
```

File: files/etc/uci-defaults/99-nethsec-default-distfeed [hunk 5]
```diff
@@ -0,0 +1,6 @@
+[ "$(uci -q get ns-plug.config.repository_url)" != "" ] && exit 0
+
+uci -q batch << EOI
+set ns-plug.config.repository_url="https://updates.nethsecurity.nethserver.org"
+commit ns-plug
+EOI
```
Comment (gsanchietti) [inline files/etc/uci-defaults/99-nethsec-default-distfeed:6]: Move the implementation inside [20_ns-plug script](https://github.com/NethServer/nethsecurity/blob/main/packages/ns-plug/files/20_ns-plug), before line 9:
- the url must be set even when updating with packages, the 20_ns-plug is [already executed on update](https://github.com/NethServer/nethsecurity/blob/main/packages/ns-plug/Makefile#L40-L48)
- the url must be set before invoking distfeed

**Review by gsanchietti — CHANGES_REQUESTED** [review]

**Review by gsanchietti — APPROVED** [review]

File: packages/ns-plug/files/20_ns-plug [hunk 6]
```diff
@@ -5,5 +5,17 @@ crontab -l | grep -q '/usr/sbin/send-backup' || echo '02 2 * * * sleep $(( RANDO
...
+if [ "$(uci -q get ns-plug.config.repository_url)" = "" ]; then
+    if [ "$(uci -q get ns-plug.config.system_id)" != "" ]; then
+        type="$(uci -q get ns-plug.config.type)"
+        system_id="$(uci -q get ns-plug.config.system_id)"
+        secret="$(uci -q get ns-plug.config.secret)"
+        uci set ns-plug.config.repository_url="https://$system_id:$secret@distfeed.nethesis.it/repository/$type/nethsecurity"
+    else
+        uci set ns-plug.config.repository_url="https://updates.nethsecurity.nethserver.org"
+    fi
+    uci commit
```
Comment (gsanchietti) [inline packages/ns-plug/files/20_ns-plug:17]:
```suggestion
    uci commit ns-plug
```

**Review by gsanchietti — CHANGES_REQUESTED** [review]

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**